### ISSUE TITLE
test_bitcoin-qt error: "Unable to init server: Could not connect: Connection refused"

### ISSUE THREAD

**Issue by ryanofsky** [bitcoin/bitcoin#23160]
**tl;dr: Running `unset QT_QPA_PLATFORMTHEME` fixes this.**

---

I recently started seeing the following error on my system running `test_bitcoin-qt`:

```bash
$ src/qt/test/test_bitcoin-qt 
Unable to init server: Could not connect: Connection refused

(test_bitcoin-qt:173399): Gtk-WARNING **: 21:03:26.220: cannot open display: 
```

strace shows the failing connect:

```
connect(8, {sa_family=AF_UNIX, sun_path=@"/run/user/1000/broadway1.socket"}, 34) = -1 ECONNREFUSED (Connection refused)
```

gdb stack trace shows:

```bash
$ gdb -ex 'catch syscall connect' -ex run --args  src/qt/test/test_bitcoin-qt 
...
Thread 1 "b-test" hit Catchpoint 1 (call to syscall connect), 0x00007ffff6757147 in connect ()
   from /nix/store/9bh3986bpragfjmr32gay8p95k91q4gy-glibc-2.33-47/lib/libpthread.so.0
(gdb) bt
#0  0x00007ffff6757147 in connect () from /nix/store/9bh3986bpragfjmr32gay8p95k91q4gy-glibc-2.33-47/lib/libpthread.so.0
#1  0x00007fffe272b7c6 in wl_display_connect () from /nix/store/wzii64jykrwq9y4x7f7hnplyk9asd1an-wayland-1.19.0/lib/libwayland-client.so.0
#2  0x00007fffe2f75a8a in _gdk_wayland_display_open () from /nix/store/5avyrz7fbxdypa272h5pciijdxh83x4h-gtk+3-3.24.30/lib/libgdk-3.so.0
#3  0x00007fffe2f42847 in gdk_display_manager_open_display () from /nix/store/5avyrz7fbxdypa272h5pciijdxh83x4h-gtk+3-3.24.30/lib/libgdk-3.so.0
#4  0x00007fffe3207ea3 in gtk_init_check () from /nix/store/5avyrz7fbxdypa272h5pciijdxh83x4h-gtk+3-3.24.30/lib/libgtk-3.so.0
#5  0x00007fffe3207ed9 in gtk_init () from /nix/store/5avyrz7fbxdypa272h5pciijdxh83x4h-gtk+3-3.24.30/lib/libgtk-3.so.0
#6  0x00007fffe3818fa6 in GnomeSettingsPrivate::GnomeSettingsPrivate(QObject*) () from /nix/store/m6np8hp9s198ilw8fbm3wbvdrp44jw10-qgnomeplatform-0.8.0/lib/libqgnomeplatform.so
#7  0x00007fffe381a00e in GnomeSettings::palette() () from /nix/store/m6np8hp9s198ilw8fbm3wbvdrp44jw10-qgnomeplatform-0.8.0/lib/libqgnomeplatform.so
#8  0x00007ffff74ffede in QApplicationPrivate::basePalette() const () from /nix/store/w4ajfg01a0ip7r5nl21c5gs2n9mgkyzi-qtbase-5.15.2/lib/libQt5Widgets.so.5
#9  0x00007ffff6e41cd0 in QGuiApplicationPrivate::setPalette(QPalette const&) () from /nix/store/w4ajfg01a0ip7r5nl21c5gs2n9mgkyzi-qtbase-5.15.2/lib/libQt5Gui.so.5
#10 0x00007ffff6e46663 in QGuiApplicationPrivate::updatePalette() () from /nix/store/w4ajfg01a0ip7r5nl21c5gs2n9mgkyzi-qtbase-5.15.2/lib/libQt5Gui.so.5
#11 0x00007ffff6e483d1 in QGuiApplicationPrivate::init() () from /nix/store/w4ajfg01a0ip7r5nl21c5gs2n9mgkyzi-qtbase-5.15.2/lib/libQt5Gui.so.5
#12 0x00007ffff7506c09 in QApplicationPrivate::init() () from /nix/store/w4ajfg01a0ip7r5nl21c5gs2n9mgkyzi-qtbase-5.15.2/lib/libQt5Widgets.so.5
#13 0x00005555556b291d in BitcoinApplication::BitcoinApplication (this=0x7fffffff0ac0) at qt/bitcoin.cpp:218
#14 0x0000555555672a61 in main (argc=<optimized out>, argv=0x7fffffff0d10) at qt/test/test_main.cpp:80
```

Qt debug output shows:

```bash
$ QT_DEBUG_PLUGINS=1 src/qt/test/test_bitcoin-qt
...
Got keys from plugin meta data ("xdgdesktopportal", "flatpak", "snap")
QFactoryLoader::QFactoryLoader() checking directory path "/run/mount/chainssd-2/home-work/bitcoin/src/qt/test/platformthemes" ...
loaded library "/nix/store/m6np8hp9s198ilw8fbm3wbvdrp44jw10-qgnomeplatform-0.8.0/lib/qt-5.15.2/plugins/platformthemes/libqgnomeplatformtheme.so"
Unable to init server: Could not connect: Connection refused

(test_bitcoin-qt:173628): Gtk-WARNING **: 21:08:27.982: cannot open display: 
```

So the failure happens loading gnome platform plugin, and apparently on my system this is enabled by an environment variable:

```bash
$ env | grep QPA
QT_QPA_PLATFORMTHEME=gnome
```

Running `unset QT_QPA_PLATFORMTHEME` before the test lets it run and pass. Maybe the test itself should be unsetting this variable in [`test_main.cpp`](https://github.com/bitcoin/bitcoin/blob/35a31d5f7e9cd71a210c1ed10abc9d772ff36049/src/qt/test/test_main.cpp#L68-L75), or maybe not if it is useful for `QApplicationPrivate` to be respecting local system settings. I just wanted to document this problem for now in case it affects anyone else.

**Comment by ryanofsky** [comment]
Closing this issue I think because it is more appropriate to deal with it in the test environment than in the test itself. But could consider other solutions later.

**Comment by fanquake** [comment]
cc @hebasto

**Comment by hebasto** [comment]
@ryanofsky 

May I ask you to share some details about your system: OS, desktop environment, Qt version (if non-static build)?

> So the failure happens loading gnome platform plugin, and apparently on my system this is enabled by an environment variable:
> 
> ```shell
> $ env | grep QPA
> QT_QPA_PLATFORMTHEME=gnome
> ```

Did you find out which installed package is responsible for setting `QT_QPA_PLATFORMTHEME=gnome`?
